pkg/commands/oscommands/os.go

```diff
@@ -109,13 +109,32 @@ func (c *OSCommand) Quote(message string) string {
 // AppendLineToFile adds a new line in file
 func (c *OSCommand) AppendLineToFile(filename, line string) error {
 	c.LogCommand(fmt.Sprintf("Appending '%s' to file '%s'", line, filename), false)
-	f, err := os.OpenFile(filename, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0o600)
+	f, err := os.OpenFile(filename, os.O_APPEND|os.O_RDWR|os.O_CREATE, 0o600)
 	if err != nil {
 		return utils.WrapError(err)
 	}
 	defer f.Close()
 
-	_, err = f.WriteString("\n" + line)
+	info, err := os.Stat(filename)
```

Comment:
Given that the logic here is a little complex, what are your thoughts on adding a test? I've found this testing package which lets you go and add files to an in-memory file system fairly easily: https://github.com/spf13/afero

We'd just have to swap out any place we use `os.<file-system method>` in this file with a call to `c.os.<file-system method>`, where by default the os field on our struct is just the default os, and from our test we can swap it out for our afero file system, similar to what we do with `getenvFn`.

Reply:
Why do we need an in memory file system for this, though? `.gitignore` _is_ a real file.

Reply:
I mean for the sake of testing. We could create a real file as part of the test, it's just that it would slow the test suite down slightly (not claiming that to be a big deal)

Reply:
> it's just that it would slow the test suite down slightly

But only a smidge? Like any other test?

I vote for a real file because... **shudders** _an additional dependency_.

Reply:
works for me

Reply:
Great! I'll add the tests then, sooner or later.